roles/upspin/tasks/service.yml

```diff
@@ -0,0 +1,42 @@
+---
+# Config Upspin system services
+
+- hosts: upspin
+
+  tasks:
+
+    - name: Load Upspin config varialbles
+      include_vars:
+        file: ../vars/config_vars.yml
+        name: config
+
+    - name: Copy local binary file to remote
+      copy:
+        src: ~/go/src/upspinserver
+        dest: $HOME/upspinserver
+        mode: preserve
+
+    - name: System service configuration
+      shell: |
+        echo "[Unit]
+        Description=Upspin server
+
+        [Service]
+        ExecStart=/home/ubuntu/upspinserver
+        User=ubuntu
+        Group=ubuntu
+        Restart=on-failure
+
+        [Install]
+        WantedBy=multi-user.target" > /etc/systemd/system/upspinserver.service
+        setcap cap_net_bind_service=+ep /home/ubuntu/upspinserver
```

Comment:
This doesn't look right.  Take a look at this again: https://upspin.io/doc/server_setup_ubuntu.md in the section _Allow upspinserver to listen on ports 80 and 443_

Reply:
I've tried to run this command as root, but the bug still occurs. Did I miss something?

Reply:
I misread this code originally and didn't see that you were running the setcap command. (I thought it was included as part of the systemd module). 

Have you looked at the logs as specified in the link above using `journalctl`? I am guessing the problem is related to the Let's Encrypt certs but I'm not certain. We need to try to narrow it down

A few suggestions

- check to make sure the systemd service is started (I think you recently changed that code)
- make sure the firewall is open on ports 80 and 443 so the server can listen for let's encrypt
- see if you can see anything inthe journalctl logs about let's encrypt
- use ansible verbose logging to understand the connection issues

Reply:
Based on your suggestions:
* From the output of Ansible verbose logging, it shows that, after I run service.yml, the state of `upspinserver` is `started`.
* Our firewall is open on ports 80 and 443.
* There are logs about let's encrypt, but everything seems doing well, until an handshake error occurred
  ```
  Mar 17 21:25:24 upspinserver[1567]: 2020/03/17 21:25:24.153886 Configuration file not found. Running in setup mode.
  Mar 17 21:25:24 upspinserver[1567]: 2020/03/17 21:25:24.153901 https: serving HTTPS on ":443" using Let's Encrypt certificates
  Mar 17 21:25:24 upspinserver[1567]: 2020/03/17 21:25:24.153906 https: caching Let's Encrypt certificates in /home/ubuntu/upspin/letsencrypt
  Mar 17 21:26:01 upspinserver[1567]: 2020/03/17 21:26:01.113493 http: TLS handshake error from 24.34.211.233:55057: read tcp 10.0.0.10:443->24.34.211.233:55057: i/o timeout
  ```
* There are no connection issues reported when running `start.yml`, even if I use `-vvvvv`. Some SSH errors DO exist in logs when running `service.yml`, but the service is started after running, so I am not sure if these logs are related.

Reply:
We used a `main.yml` to run all the playbooks.
What really confused me is that, everytime we run the `main.yml`, it fails in the last step, which is the task in `start.yml`, the command is `upspin setupserver`.
But when I run them separately, or run `start.yml` again after finishing `main.yml`, it works perfectly well and the Upspin server is ready to use.
I do not know if it is an issue of how Ansible run tasks on different hosts, because we need to do server configuration on our remote server first, after that, we can successfully run `upspin setupserver` on our local machine.
The order of tasks should be as same as in `main.yml`, I also tried `--fork=1` option to ensure there is no parallel execution on different hosts, but still doesn't work. Do you think it is a problem of running order?